core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,348 @@
+        self.assigned_reviewer_id = assigned_reviewer_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = exp_domain.ExplorationChange(change_cmd)
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type SuggestionEditStateContent.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                SuggestionEditStateContent object are invalid.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if not isinstance(self.change_cmd, exp_domain.ExplorationChange):
+            raise utils.ValidationError(
+                'Expected change_cmd to be an ExplorationChange, received %s'
+                % type(self.change_cmd))
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_CONTENT:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_CONTENT,
+                    self.get_score_type()))
+
+        if self.get_score_sub_type() not in constants.ALL_CATEGORIES:
+            raise utils.ValidationError(
+                'Expected the second part of score_category to be a valid'
+                ' category, received %s' % self.get_score_sub_type())
+
+        if self.change_cmd.cmd != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, received %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd.cmd))
+
+        if (self.change_cmd.property_name !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, received %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd.property_name))
+
+    def pre_accept_validate(self):
+        """Performs referential validation. This function needs to be called
+        before accepting the suggestion.
+        """
+        self.validate()
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd.state_name not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
+                self.change_cmd.state_name)
+
+    def get_change_list_for_accepting_suggestion(self):
+        """Gets a complete change_cmd for the suggestion.
+
+        Returns:
+            list(ExplorationChange). The change_list corresponding to the
+                suggestion.
+        """
+        change_cmd = self.change_cmd
+        exploration = exp_services.get_exploration_by_id(self.target_id)
+        old_content = (
+            exploration.states[self.change_cmd.state_name].content.to_dict())
+
+        change_cmd.old_value = old_content
+        change_cmd.new_value['audio_translations'] = (
```

Comment:
One thought: if the content changes, wouldn't the audio translations need to be marked stale in some cases? Maybe you'll want the reviewer to indicate via a checkbox whether this is a "big" change or not, and act accordingly here.

Could be a separate PR, but make sure to track it.

Reply:
Oh true. So the first thought was to pass in an extra parameter to accept_suggestion which says whether or not to mark these translations as needing update. However this parameter would not make sense in some suggestion scenarios which do not involve translations (like say add question to topic). So is it ok to pass in None in those situations (I don't think this affects the generality too much though).

Also yeah. I think I'll push it in a separate PR (as this blocks the next PR for this milestone).

Reply:
It's unclear to me whether this should be considered a modification of the changelist, or whether the changelist should come from the editor frontend, or something else. I lean towards finishing up the e2e experience for now and then adding this in as a feature after that (please file an issue).

Reply:
(Also feel free to merge if you're happy with this :) )

Reply:
Yep. Will continue further tasks and complete the full experience then decide what's appropriate here. Will file an issue. 

Merging thanks!